cpp/include/cuspatial/utility/geometry_utility.cuh

```diff
@@ -0,0 +1,96 @@
+#pragma once
```

Comment:
The path "utility/geometry_utility..." is a bit redundant. Also, I fear that "geometry utilities" could grow quite large. All of these are linestring / segment utilities, so perhaps a more specific filename is in order? 

Also these are all detail namespace so shouldn't `detail` be in the path? 

e.g. `include/cuspatial/detail/linestring_utilities.cuh` 

Looking at the utility folder, I think that `vec_2d.cuh` should be promoted to the top-level include directory (since it is not detail -- it's analogous to `constants.hpp`). Everything else in `utility` should move to `detail`. Thoughts?

Reply:
Great advice. In the other PR you mentioned we can move `device_atomics.cuh` to either within a `utility` folder or not. I think most methods here all fits under the utility category so should probably create a folder for that.

Reply:
Moved as noted except `device_atomics.cuh` will be handled in the other PR.